src/ledger/LedgerManagerImpl.cpp

```diff
@@ -582,6 +595,7 @@ LedgerManagerImpl::closeLedger(LedgerCloseData const& ledgerData)
     std::unique_ptr<LedgerCloseMeta> ledgerCloseMeta;
     if (mMetaStream)
     {
+        emitAnyStoredSafeMetaAfterLedgerValidation();
```

Comment:
How does this work on the first ledger after restart? Imagine that ledger 73 is closed, then the node restarts before ledger 74 is closed. What happens to the meta for ledger 73? I don't think you are serializing it anywhere, so it seems to be gone. Am I missing something? @MonsieurNicolas initially pointed this out as a potential pitfall in https://github.com/stellar/stellar-core/issues/2925#issuecomment-805398657 so I want to make sure we're handling it.

Reply:
This is what I think happens: https://github.com/stellar/stellar-core/issues/2925#issuecomment-807575702 , but I'll re-check.

Reply:
> > How does this work on the first ledger after restart? Imagine that ledger 73 is closed, then the node restarts before ledger 74 is closed. What happens to the meta for ledger 73? I don't think you are serializing it anywhere, so it seems to be gone. Am I missing something?

> This is what I think happens: #2925 (comment) , but I'll re-check.

Although actually, I don't think Horizon's specific current policy is really relevant.  Captive Core's stateless (hence the requirement for the in-memory DB in the config); what happens after a restart is that Horizon has to ask `stellar-core` to `--start-from` at least as early as the first one that Horizon hasn't already ingested, regardless of what ledger `stellar-core` had last closed at the time of the restart, or what ledger the network is on when the node comes back up.  Even before this change, it's possible for the core to have closed later ledgers when Horizon hasn't ingested earlier ones -- Core's just dumping stuff to a pipe; nothing guarantees that Horizon has consumed it and gotten it all persisted in its own database before Core closes more ledgers.  When there's a restart, Horizon has to look at its database, see what ledger it last ingested, and ask Core to start streaming with the next one, which Core will do.  The difference with this change is that if Horizon asks it to start streaming ledger `N`, then Core will give it the meta from ledger `N` when ledger `N+1` closes, not when ledger `N` closes.  But that doesn't mean it will _skip_ `N`; it just delivers its meta a ledger late, as with all ledgers that it closes.